docs/source/release_notes/release-notes.rst

```diff
@@ -56,6 +56,13 @@ develop
     *    - |devbreak|
          - IteratorUtils.forEach removed; it's not needed in a Java 8 codebase.
            (`Pull Request <https://github.com/palantir/atlasdb/pull/2207>`__)
+           
+    *    - |improved|
+         - Timelock now creates client namespaces the first time they are requested, rather than requiring them to be specified in config.
+           This means that specifying a list of clients in Timelock configuration will no longer have any effect. Further, a new configuration property called ``max-number-of-clients`` has been introduced in ``TimeLockRuntimeConfiguration``. This can be used to limit the number of clients that will be created dynamically, since each distinct client has some memory, disk space, and CPU overhead.
```

Comment:
yeah, i was thinking about the potential overheads of actually creating too many clients.

probably prudent to do some quick IL testing to confirm, but the biggest thing is probably just creating the paxos directory. The overhead of an empty timestamp and lock services is minimal; the only thing i think that would use any CPU is the lock reaper which just runs once per second